File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/robots/bobobot/Robot.java
package org.firstinspires.ftc.teamcode.robots.bobobot;

import com.qualcomm.robotcore.hardware.HardwareMap;

import org.firstinspires.ftc.robotcore.external.Telemetry;


public class Robot {
        Telemetry telemetry;
        DriveTrain driveTrain;

        public Robot(Telemetry telemetry, HardwareMap hardwareMap)
        {
            this.telemetry=telemetry;
            driveTrain = new DriveTrain(telemetry, hardwareMap);

            motorInit();
        }
        public void motorInit()
        {
            driveTrain.motorInit();

        }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/powerplay/Robot.kt
package org.firstinspires.ftc.teamcode.powerplay

import com.qualcomm.robotcore.hardware.HardwareMap
import org.firstinspires.ftc.teamcode.powerplay.subsystems.Claw
import org.firstinspires.ftc.teamcode.powerplay.subsystems.Slides
import org.firstinspires.ftc.teamcode.util.Subsystem

class Robot: Subsystem {
    private val claw = Claw()
    private val slides = Slides()
    private val subsystems = mutableListOf(
        claw,slides
    )

    fun clawOpened() {
        claw.openClaw()
    }

    fun clawClosed() {
        claw.closeClaw()
    }

    fun slidesUp() {
        slides.up()
    }

    fun slidesDown() {
        slides.down()
    }

    fun slidesOff() {
        slides.off()
    }

    override fun init(hardwareMap: HardwareMap) {
        subsystems.forEach{it.init(hardwareMap)}
    }

    override fun update() {
        subsystems.forEach{it.update()}
    }

    override fun reset() {
        subsystems.forEach{it.reset()}
    }
}